mllib/src/main/scala/org/apache/spark/ml/source/libsvm/LibSVMRelation.scala

```diff
@@ -89,18 +93,17 @@ private[libsvm] class LibSVMFileFormat extends TextBasedFileFormat with DataSour
       files: Seq[FileStatus]): Option[StructType] = {
     val libSVMOptions = new LibSVMOptions(options)
     val numFeatures: Int = libSVMOptions.numFeatures.getOrElse {
-      // Infers number of features if the user doesn't specify (a valid) one.
-      val dataFiles = files.filterNot(_.getPath.getName startsWith "_")
```

Comment:
I am quite sure that we can remove this - https://github.com/apache/spark/commit/76622c661fcae81eb0352c61f54a2e9e21a4fb98.